### PR TITLE
fix(ui5-datepicker): display extreme values correctly

### PR DESCRIPTION
**Pull request by a-z-ivanov**
Fixes https://github.com/SAP/ui5-webcomponents/issues/26

**Thank you for your contribution!** 👏

To get it merged faster, kindly review the checklist below:

## Pull Request Checklist
- [ ] Reviewed the [Contributing Guidelines](https://github.com/SAP/ui5-webcomponents/blob/master/CONTRIBUTING.md)
    + Especially the [How to Contribute](https://github.com/SAP/ui5-webcomponents/blob/master/CONTRIBUTING.md#how-to-contribute) section 
- [ ] [Correct commit message style](https://github.com/SAP/ui5-webcomponents/blob/master/docs/Guidelines.md#commit-message-style)

### CODE REVIEW

File: packages/main/src/Calendar.js [hunk 1]
```diff
@@ -356,36 +362,66 @@ class Calendar extends WebComponent {
...
-			// add 20 years to the timestamp of the monthpicker
-			timestamp: this._yearPicker.timestamp + (31536000 * 20),
+			timestamp: this._yearPicker.timestamp + (31536000 * YearPicker._ITEMS_COUNT),
 		});
 
 		this._isShiftingYears = true;
 	}
 
 	_showPrevPageYears() {
+		if (this._yearPicker.timestamp) {
```
Comment (ilhan007) [inline packages/main/src/Calendar.js]: These blocks (the one highlighted here) and above seem very similar,  perhaps you can extract and reuse code.

File: packages/main/src/DatePicker.js [hunk 2]
```diff
@@ -464,6 +464,11 @@ class DatePicker extends WebComponent {
 	}
 
 	_changeCalendarSelection() {
+		if (this._calendarDate.getYear() < 1) {
+			// 0 is a valid year, but we cannot display it
+			return;
+		}
```
Comment (ilhan007) [inline packages/main/src/DatePicker.js]: discussed, as 0 is valid lets display it.

Comment (a-z-ivanov) [reply packages/main/src/DatePicker.js]: there is a problem in islamic. 1 is the minimum for now

File: packages/main/src/Calendar.js (hunk 1 above)
Comment (a-z-ivanov) [reply packages/main/src/Calendar.js]: done

**Pull request merged**